packages/fuel-gauge/src/funding-transaction.test.ts

```diff
@@ -232,4 +236,109 @@ describe(__filename, () => {
 
     expect(getResourcesToSpend).toHaveBeenCalledTimes(2);
   });
+
+  it('should ensure a partially funded Transaction will require only missing funds', async () => {
+    const receiver = Wallet.generate({ provider });
+    const wallet1 = Wallet.generate({ provider });
+    const wallet2 = Wallet.generate({ provider });
+
+    const totalInBaseAsset = 15_000;
+    const totalInAssetA = 20_000;
+    const partiallyInAssetA = totalInAssetA / 2;
+
+    // wallet1 does not have enough funds to cover total required in assetA and
```

Comment:
Not understanding this comment

Reply:
@maschad is it more clear now? https://github.com/FuelLabs/fuels-ts/pull/2267/commits/bb25beffc6021b04462596c4d88ccad695b6c998